msrest/serialization.py

```diff
@@ -1486,6 +1486,25 @@ def _classify_target(self, target, data):
             pass  # Target is not a Model, no classify
         return target, target.__class__.__name__
 
+    def failsafe_deserialize(self, target_obj, response_data, content_type=None):
+        """Ignores any errors encountered in deserialization,
+        and falls back to not deserializing the object. Recommended
+        for use in error deserialization, as we want to return the
+        HttpResponseError to users, and not have them deal with
+        a deserialization error.
+
+        :param str target_obj: The target object type to deserialize to.
+        :param str/dict data: The response data to deseralize.
+        """
+        try:
+            data = self._unpack_content(response_data, content_type)
```

Comment:
added this content unpacking line, since `call` does content unpacking of the data as well, and as this method doesn't enter `__call__`, added this code here